### PR TITLE
Make xtasks be able to use the specified binary

### PR DESCRIPTION
**Pull request by Kerollmops**
Makes it possible to specify the binary to run. It is useful to run PGO optimized binaries.

### CODE REVIEW

File: crates/xtask/src/bench/mod.rs [hunk 1]
```diff
@@ -86,6 +86,10 @@ pub struct BenchDeriveArgs {
     /// The maximum time in seconds we allow for fetching the task queue before timing out.
     #[arg(long, default_value_t = 60)]
     tasks_queue_timeout_secs: u64,
+
+    /// The path to the binary to run. By default it compiles the binary with cargo.
```
Comment (dureuill) [inline crates/xtask/src/bench/mod.rs:90]:
```suggestion
    /// The path to the binary to run. 
    /// 
    /// If unspecified, runs `cargo run` after building Meilisearch with `cargo build`
```

File: crates/meilisearch/src/main.rs [hunk 2]
```diff
@@ -129,6 +129,11 @@ async fn try_main() -> anyhow::Result<()> {
 
     print_launch_resume(&opt, analytics.clone(), config_read_from);
 
+    tokio::spawn(async move {
+        tokio::signal::ctrl_c().await.unwrap();
+        std::process::exit(77);
```
Comment (dureuill) [inline crates/meilisearch/src/main.rs:134]:
```suggestion
        std::process::exit(130);
```

**Review by dureuill — APPROVED** [review]
Thank you, see inline comments

**Comment by dureuill** [comment]
(also we need to run rustfmt)

**Comment by Kerollmops** [comment]
> The PR looks good. I'm just wondering: do we want to do such a significant change as starting to handle the SIGTERM signal so close of v1.12 release?

I think that's it is a really small change compared to all of the new indexer. We can afford this one. We also need it for future development (PGO) anyway.

**Review by dureuill — APPROVED** [review]
as I said the code looks good to me 😊

**Comment by Kerollmops** [comment]
bors merge

**Comment by dureuill** [comment]
bors merge

**Pull request merged**